addons/binding/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/handler/MystromBridgeHandler.java

```diff
@@ -0,0 +1,290 @@
+import com.google.gson.JsonParser;
+
+/**
+ * This bridge handler connects to mystrom and handles all the api requests. It pulls down the
+ * updated data, polls the system and does all the co-ordination with the other handlers
+ * to get the data updated to the correct things.
+ *
+ * @author Stéphane Raemy - initial contribution
+ */
+public class MystromBridgeHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(MystromBridgeHandler.class);
+
+    // REST Client API variables
+    protected final Client mystromClient = ClientBuilder.newClient();
+    public final WebTarget mystromTarget = mystromClient.target(MYSTROM_URI);
+    public final WebTarget authTarget = mystromTarget.path(MYSTROM_AUTH_PATH);
+    public final WebTarget devicesTarget = mystromTarget.path(MYSTROM_DEVICES_PATH);
+    public final WebTarget deviceTarget = mystromTarget.path(MYSTROM_DEVICE_PATH);
+    public final WebTarget deviceSwitchTarget = mystromTarget.path(MYSTROM_DEVICE_SWITCH_PATH);
+
+    private List<MystromDeviceAddedListener> listeners = new ArrayList<MystromDeviceAddedListener>();
+
+    // Will refresh the data each time it runs.
+    private Runnable pollingRunnable = new Runnable() {
+        @Override
+        public void run() {
+            refreshData();
+        }
+    };
+
+    private ScheduledFuture<?> pollingJob;
+
+    protected Gson gson = new Gson();
+
+    /**
+     * Creates the bridge handler to connect to mystrom.
+     *
+     * @param bridge The bridge to connect to mystrom with.
+     * @param bindingProperties
+     */
+    public MystromBridgeHandler(@NonNull Bridge bridge) {
+        super(bridge);
+    }
+
+    /**
+     * Initialize the connection to mystrom.
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initialize the Mystrom bridge handler");
+
+        MystromBridgeConfiguration config = getConfigAs(MystromBridgeConfiguration.class);
+        logger.debug("Email       {}.", config.email);
+        logger.debug("Password   {}.", config.password);
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Starting poll query");
+
+        startAutomaticRefresh(config.refreshInterval);
+    }
+
+    /**
+     * Do something useful when the configuration update happens. Triggers changing
+     * polling intervals as well as re-doing the access token.
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        logger.debug("Config update");
+        super.handleConfigurationUpdate(configurationParameters);
+
+        stopAutomaticRefresh();
+        startAutomaticRefresh(getConfigAs(MystromBridgeConfiguration.class).refreshInterval);
+    }
+
+    /**
+     * Clean up the handler.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Mystrom bridge disposed");
+        stopAutomaticRefresh();
+        this.pollingJob = null;
+        this.pollingRunnable = null;
```

Comment:
This will remove the runnable, but is never set again, so after an configuration update this method is called, and then the class is initialized again butt pollingRunnable is null.